staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go

```diff
@@ -537,18 +537,35 @@ func (b *builder) getOpenAPIConfig(v2 bool) *common.Config {
 			buildDefinitions.Do(generateBuildDefinitionsFunc)
 			return namer.GetDefinitionName(name)
 		},
-		GetDefinitions: func(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
-			def := utilopenapi.GetOpenAPIDefinitionsWithoutDisabledFeatures(generatedopenapi.GetOpenAPIDefinitions)(ref)
```

Comment:
I believe the initial reason for not using the precomputed definitions map was the passed in ref function here.

Your code switches on v2/v3 so I think it is correct, just something to be aware of

Reply:
Right I think it's also because we don't want to modify the map (as it's done just below). Separating into two lists: one "Definition" and one "Additional" solves that problem.